src/router.ts

```diff
@@ -0,0 +1,258 @@
+
+        // Increase production to dump container that already feeds node relay, if possible
+        for (const dumpRoute of node.dumpRoutes) {
+            if (dumpRoute.relayRoutes.includes(relayRoute)) {
+                // How much overproduction we currently have
+                let currentSurplus =
+                    dumpRoute.container.ingress(node.item) - dumpRoute.container.egress(node.item)
+
+                // How much more ingress can we produce
+                const newIngress = dumpRoute.container.incomingLinksFree * node.rate
+
+                // New industries we require
+                let newEgress = Math.min(newIngress + currentSurplus, relayEgress)
+
+                // New industries we require
+                const newIndustries = Math.min(
+                    Math.ceil((newEgress - currentSurplus) / node.rate),
+                    dumpRoute.container.incomingLinksFree,
+                )
+                newEgress = currentSurplus + newIndustries * node.rate
+                for (let i = 0; i < newIndustries; i++) {
+                    const industry = node.factory.createIndustry(node.item, dumpRoute.container)
+                    dumpRoute.industries.push(industry)
+                }
+
+                relayRoute.transferUnit.increaseTransferRate(dumpRoute.container, newEgress)
+            }
+            relayEgress =
+                relayRoute.container.egress(node.item) - relayRoute.container.ingress(node.item)
+        }
+        if (relayEgress <= delta) {
+            continue
+        }
+    }
+
+    // Try to satisfy all relays using existing dump containers
+    for (const relayRoute of node.getRelayRoutes()) {
+        // Unsatisfied container egress
+        let relayEgress =
+            relayRoute.container.egress(node.item) - relayRoute.container.ingress(node.item)
+        if (relayEgress <= delta) {
+            continue
+        }
+
+        // Add to existing route if possible
+        for (const dumpRoute of node.dumpRoutes) {
+            // skip if this dump route is full
+            if (!dumpRoute.container.canAddOutgoingLinks(1)) {
+                continue
+            }
+
+            // if node dump route is feeding a relay that is supplied by more than
+            // one dump route, then we cannot add any new routes to node dump container
+            let splitDump = false
+            for (const checkRelayRoute of dumpRoute.relayRoutes) {
+                for (const checkDumpRoute of node.dumpRoutes) {
+                    if (checkDumpRoute === dumpRoute) {
+                        continue
+                    }
+                    if (checkDumpRoute.relayRoutes.includes(checkRelayRoute)) {
+                        splitDump = true
+                        break
+                    }
+                }
+                if (splitDump) {
+                    break
+                }
+            }
+            if (splitDump) {
+                continue
+            }
```

Comment:
```suggestion
            const splitDump = dumpRoute.relayRoutes.some(checkRelayRoute =>
                node.dumpRoutes.some(checkDumpRoute => 
                    checkDumpRoute !== dumpRoute && checkDumpRoute.relayRoutes.includes(checkRelayRoute))
            )
            if (splitDump) {
                continue
            }
```
Maybe we can avoid some of those `break`s and `continue`s. 😄